etl/steps/data/garden/iea/2024-07-04/critical_minerals.py

```diff
@@ -0,0 +1,359 @@
+    # NOTE: Only include it if the difference is larger than 1%.
+    tb_other_low = (
+        combined[combined["difference"] >= (combined["demand"] * 0.01)]
+        .assign(**{"technology": OTHER_LOW_LABEL})
+        .drop(columns=["demand", "demand_summary"])
+        .rename(columns={"difference": "demand"})
+        .reset_index(drop=True)
+    )
+
+    # Append the demand from other low emissions power generation to the demand table.
+    tb_demand_with_other = pr.concat([tb_demand, tb_other_low], ignore_index=True)
+
+    return tb_demand_with_other
+
+
+def add_demand_from_other_graphite(tb_demand: Table, tb_total: Table) -> Table:
+    # Calculate "Graphite, other uses", which should be the total from sheet 1 minus the total from sheets 4.x for each mineral. Add those other uses to the demand table.
+
+    # List all graphite-related minerals listed in the 4.x sheets.
+    graphite_uses = ["Battery-grade graphite"]
+
+    # Sanity checks.
+    error = "There are new unexpected uses of graphite in the 4.x sheets."
+    assert set(tb_demand[tb_demand["mineral"].str.lower().str.contains("graphite")]["mineral"]) == set(
+        graphite_uses
+    ), error
+    error = "There are new unexpected uses of graphite in Sheet 1."
+    assert set(tb_total[tb_total["mineral"].str.lower().str.contains("graphite")]["mineral"]) == set(
+        [GRAPHITE_ALL_LABEL]
+    ), error
+
+    # Calculate the total graphite demand in the total demand sheet.
+    tb_graphite_total = (
+        tb_total[tb_total["mineral"] == GRAPHITE_ALL_LABEL].drop(columns=["mineral"]).reset_index(drop=True)
+    )
+
+    # For now we only want to add the "other graphite uses" to each individual technology.
+    # But the "Other uses" will be added to all minerals later.
+    # Hence, remove other uses and total demand.
+    tb_graphite_total = tb_graphite_total[
+        ~tb_graphite_total["technology"].isin([TOTAL_DEMAND_LABEL, OTHER_LABEL])
+    ].reset_index(drop=True)
+
+    # Calculate the total graphite demand in the 4.x sheets.
+    tb_graphite_demand = (
+        tb_demand[tb_demand["mineral"].isin(graphite_uses)].drop(columns=["mineral"]).reset_index(drop=True)
+    )
+    combined = tb_graphite_demand.merge(
+        tb_graphite_total, how="inner", on=["case", "scenario", "technology", "year"], suffixes=("", "_summary")
+    )
+    # Check that the totals in sheet 1 are always larger than the totals calculated from combining the 4.x sheets (or within 1%).
+    error = "Sheet on clean technology graphite demand by mineral should be >= the sum of the graphite demands from the 4.x sheets."
+    assert combined[(combined["demand_summary"] < (combined["demand"] * 0.99))].empty, error
+    combined["difference"] = combined["demand_summary"] - combined["demand"]
+
+    # Create a new technology that contains all the remaining graphite demand not included in the 4.x sheets.
+    # NOTE: Only include it if the difference is larger than 1%.
+    tb_graphite_other = (
```

Comment:
I agree that a description processing would be expected, but I'm not sure about what its content should be. Most of the choices are not so much about what we do to the data, but how we put together data points from the different sheets, given the very inconvenient format. The graphite case is the one where there is more processing, but explaining this would be tricky ("In this sheet, ..., but in the other sheet, ..."). It may take a fairly long description and not be very informative for any user that has not opened the file.

Reply:
If the graphite data ends up being used, I think the best solution would be to contact the data provider, because this may actually be a data issue (given that for all other minerals this inconsistency does not happen).